de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/user/UserRepository.java

```diff
@@ -40,9 +39,14 @@
 @Repository
 public class UserRepository
 {
-	public User getById(@NonNull final UserId userId)
+	public User getByIdInTrx(@NonNull final UserId userId)
 	{
-		final I_AD_User userRecord = loadOutOfTrx(userId.getRepoId(), I_AD_User.class);
+		final I_AD_User userRecord = load(userId.getRepoId(), I_AD_User.class);
+
+		if(userRecord == null)
```

Comment:
i think we can replace that `if null` with some `Check.assumeNotNull`.
Ideally would be to change the `load` method and make sure it does not return `null`.... but for that we need to measure the impact first..... so, not now.

Reply:
Done